core/src/main/java/org/apache/iceberg/MetricsConfig.java

```diff
@@ -104,38 +108,40 @@ public static MetricsConfig forTable(Table table) {
    * @param table an Iceberg table
    */
   public static MetricsConfig forPositionDelete(Table table) {
-    MetricsConfig config = new MetricsConfig();
+    ImmutableMap.Builder<String, MetricsMode> columnModes = ImmutableMap.builder();
 
-    config.columnModes.put(MetadataColumns.DELETE_FILE_PATH.name(), MetricsModes.Full.get());
-    config.columnModes.put(MetadataColumns.DELETE_FILE_POS.name(), MetricsModes.Full.get());
+    columnModes.put(MetadataColumns.DELETE_FILE_PATH.name(), MetricsModes.Full.get());
+    columnModes.put(MetadataColumns.DELETE_FILE_POS.name(), MetricsModes.Full.get());
 
     MetricsConfig tableConfig = forTable(table);
 
-    config.defaultMode = tableConfig.defaultMode;
+    MetricsMode defaultMode = tableConfig.defaultMode;
     tableConfig.columnModes.forEach((columnAlias, mode) -> {
       String positionDeleteColumnAlias = DOT.join(MetadataColumns.DELETE_FILE_ROW_FIELD_NAME, columnAlias);
-      config.columnModes.put(positionDeleteColumnAlias, mode);
+      columnModes.put(positionDeleteColumnAlias, mode);
     });
 
-    return config;
+    return new MetricsConfig(columnModes.build(), defaultMode);
   }
 
   private static MetricsConfig from(Map<String, String> props, SortOrder order) {
-    MetricsConfig spec = new MetricsConfig();
+    Map<String, MetricsMode> columnModes = new HashMap<>();
```

Comment:
Why not use the ImmutableMap builder like previous method?

Reply:
+1, and we prefer `Maps.newHashMap` over raw constructor for hash map.

Reply:
we cannot use ImmutableMap builder, because we write and then overwrite certain keys in the map

Reply:
in that case could you move to `Maps.newHashMap`?